dbschema/partnership-producing-medium.esdl

```diff
@@ -0,0 +1,7 @@
+module default {
+  type PartnershipProducingMedium extending Resource {
+    required medium: Product::Medium;
+
+    partnership: Partnership;
+  }
```

Comment:
I'm unsure if there is another connection here.  The `UpdatePartnershipProducingMediumOutput` contains an `engagementId`.  Is there an `Engagement` connection as well?

Reply:
Yes. This could probably be an `Engagement::Child`.
Also
```esdl
constraint exclusive on ((.engagement, .partnership, .medium));
```
I'd make `partnership` required too.

The lists from API include all mediums for the engagement, so if the record is not found, then the partnership is null. But that's getting into query territory.

Reply:
For the `Engagement::Child` comment above...do you mean this type should extend `Engagement::Child`, or there should be a new property called `engagement` of that type?

Reply:
extend. This _is_ an engagement child.